file/file_prefetch_buffer.cc

```diff
@@ -23,18 +24,16 @@ namespace ROCKSDB_NAMESPACE {
 
 void FilePrefetchBuffer::CalculateOffsetAndLen(size_t alignment,
                                                uint64_t offset,
-                                               size_t roundup_len, size_t index,
-                                               bool refit_tail,
+                                               size_t roundup_len,
+                                               uint32_t index, bool refit_tail,
                                                uint64_t& chunk_len) {
   uint64_t chunk_offset_in_buffer = 0;
   bool copy_data_to_new_buffer = false;
   // Check if requested bytes are in the existing buffer_.
   // If only a few bytes exist -- reuse them & read only what is really needed.
   //     This is typically the case of incremental reading of data.
   // If no bytes exist in buffer -- full pread.
-  if (bufs_[index].buffer_.CurrentSize() > 0 &&
-      offset >= bufs_[index].offset_ &&
-      offset <= bufs_[index].offset_ + bufs_[index].buffer_.CurrentSize()) {
```

Comment:
The check here is ```offset <= bufs_[index].offset_ ...```, whereas its ```offset < bufs_[index].offset_ ...``` in ```IsOffsetInBuffer```. I assume the latter is correct, but just want to check.

Reply:
Yes, latter one is correct.